apps/sms/services/js/moz_mobile_message/moz_mobile_message_shim.js

```diff
@@ -0,0 +1,231 @@
+/* -*- Mode: js; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- /
+/* vim: set shiftwidth=2 tabstop=2 autoindent cindent expandtab: */
+
+
+/* global bridge */
+
+'use strict';
+(function() {
+
+/**
+ * Name of the service for mozMobileMessage API shim
+ * @type {string}
+ */
+
+const SERVICE_NAME = 'mozMobileMessageShim';
+
+/**
+ * Array of the event name that is corresponding to mozMobileMessage events.
+ * @type {Array}
+ */
+
+const EVENTS = ['received', 'sending', 'sent', 'failed', 'deleted',
+                'readsuccess', 'deliverysuccess'];
+
+/**
+ * Array of method that need to be exposed for API shim.
+ * @type {Array}
+ */
+
+const METHODS = ['getMessage', 'retrieveMMS', 'send', 'sendMMS',
+                 'delete', 'markMessageRead', 'getSegmentInfoForText'];
+
+/**
+ * Array of stream that need to return data(messages/threads) in chunk.
+ * @type {Array}
+ */
+
+const STREAMS = ['getThreads', 'getMessages'];
+
+var mozMobileMessage = navigator.mozMobileMessage;
```

Comment:
Maybe ```const``` here as well. I'm always suggesting this, but I'm not sure if it really helps Gecko to optimize (or maybe it makes things worse currently :)), but at least it's clear that variable is not supposed to be changed :)